lib/elixir/lib/macro.ex

```diff
@@ -2686,6 +2695,29 @@ defmodule Macro do
     end
   end
 
+  # This is the last clause of a cond
+  defp dbg_cond_tree([{:->, _meta, [[left], right]}], index) do
+    quote do
+      # Keeping a cond here so to respect the CondClauseError behavior.
+      cond do
+        clause_value = unquote(left) ->
+          {unquote(escape(left)), clause_value, unquote(index), unquote(right)}
+      end
+    end
+  end
+
+  defp dbg_cond_tree([{:->, _meta, [[left], right]} | rest], index) do
+    cond_tree_ast = dbg_cond_tree(rest, index + 1)
```

Comment:
Apologies but why do we rewrite to an `if`? Couldn't we keep it as a `cond`?

Reply:
Sorry, I was overthinking it. We can just transform the clauses like we did for `case`, looks much simpler:
[4044eb9](https://github.com/elixir-lang/elixir/pull/12475/commits/4044eb918cf671cc6cd2b14284d06af4f84d75e1)